client/submitAssignment.js

```diff
@@ -0,0 +1,65 @@
+
+$("#myForm").submit(function(event) {
+    var formData = new FormData(this);
+    event.stopPropagation();
+    event.preventDefault();
+    $.ajax({
+      url: "http://localhost:8083/upload",
+      data: formData,
+      processData: false,
+      contentType: false,
+      type: 'POST',
+	  cache: false,
+      success: function(data) {
+       console.log(data);
+	   let output = document.getElementById("output");
+	   output.innerHTML = data;
+      },
+	  error: function(data) {
+       console.log(data);
+	   let output = document.getElementById("output");
+	   output.innerHTML = data;
+      }
+    });
+    return false;
+  });
+
+// function to validate the file upload and command line args
+function validateForm() {
+  var fileName = document.forms["myForm"]["file"].value;
+  if (fileName == "") {
+	let output = document.getElementById("fileError");
+	output.innerHTML = "Please select a file to upload";
+    return false;
+  }
+}
+
+// function to verify the file type (should be .zip, .tar or .tar.gz)
+function verifyFiles() {
+    let allowedFiles = [".zip", ".tar", ".tar.gz"];
+    let fileUpload = document.getElementById("file");
+    let output = document.getElementById("fileError");
+    let regex = new RegExp("([a-zA-Z0-9\s_\\.\-:()])+(" + allowedFiles.join('|') + ")$");
```

Comment:
is using regular expressions the best way to validate files? Is there a library that can be used for this?

Reply:
Yeah. There is a library for checking file type. I already have that code. But forgot to check in, this was the previous code. However, even if file type is checked using the library, it still is not able to identify if any file extension is manually changed. That thing is detected by the server and it sends a file error in such cases

Reply:
that works.